internal/app/service/tests/achievement_service_test.go

```diff
@@ -95,3 +97,85 @@ func TestCreateAchievement(t *testing.T) {
 		})
 	}
 }
+
+var (
+	profileID = "123"
+)
+
+func TestGetAchievements(t *testing.T) {
+	mockAchievementRepo := new(mocks.AchievementStorer)
+	var repodeps = service.RepoDeps{
+		AchievementDeps: mockAchievementRepo,
+	}
+	achService := service.NewServices(repodeps)
+
+	// mock data
+	mockProfileId := profileID
+	mockResponseAchievement := []dto.AchievementResponse{
+		{
+			ProfileID:   123,
+			Name:        "Client Appreciation",
+			Description: "Appreciated by client for the work done",
+		},
+	}
+
+	tests := []struct {
+		Name            string
+		ProfileID       string
+		MockSetup       func(*mocks.AchievementStorer, string)
+		isErrorExpected bool
+		wantResponse    []dto.AchievementResponse
+	}{
+		{
+			Name:      "Success",
+			ProfileID: mockProfileId,
+			MockSetup: func(mockAchievementStorer *mocks.AchievementStorer, profileID string) {
+				profileIDInt, _ := strconv.Atoi(profileID)
+				mockAchievementStorer.On("GetAchievements", mock.Anything, profileIDInt).Return(mockResponseAchievement, nil).Once()
+			},
+			isErrorExpected: false,
+			wantResponse:    mockResponseAchievement,
+		},
+		{
+			Name:      "Fail get achievement",
```

Comment:
use `fail_get_achievement` and use same pattern for other places

Reply:
kk